### PR TITLE
Create a Singularity Image Definition

### PR DESCRIPTION
**Pull request by martyngigg**
The main aim of this work is to create a Singularity image containing the frontend and backend code for Journal Viewer 2 such that it can be deployed to platforms such as IDAaaS.

During the course of this work several other things have been done:

*  Switched from Nix to Conan: The Nix flake definition was copied from Dissolve and is quite complicated for a simple application. Conan is a good alternative for C++ and very well supported by the community. As many of the dependencies as possible have been statically linked into the final binary to simplify deployment.
* Moved all CMake/C++ code to frontend directory
* Create a Python package `jv2backend` out of the Python files for easy of building and installation

The singularity definition files are in `scripts/linux` and are split into a version that builds the code (`jv2-build.def`) and another runtime image that just copied files from the build image: `jv2.def`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,71 @@
...
+to install the Python package and its dependencies to avoid polluting
+your main Python environment:
+
+```sh
+cd backend
+# create environment in local directory
+python3 -m venv ./venv
+# activate
+source ./venv/bin/activate
+python -m pip install --editable .
```
Comment (trisyoungs) [inline README.md]:
```suggestion
python3 -m pip install --editable .
```

File: backend/src/jv2backend/__init__.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2022 E. Devlin and T. Youngs
```
Comment (trisyoungs) [inline backend/src/jv2backend/__init__.py]: Missing someone here? :smile:

File: scripts/linux/jv2launcher [hunk 3]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+# Launch script to wrap starting the backend & frontend.
+# Ensures the server shutsdown when the GUI exits
```
Comment (trisyoungs) [inline scripts/linux/jv2launcher]:
```suggestion
# Ensures the server shuts down when the GUI exits
```

**Review by trisyoungs — APPROVED** [review]
Comprehensive, and looking an awful lot cleaner in a lot of places!
Couple of small typos noted in the comments but otherwise ready to go.

**Comment by martyngigg** [comment]
Thanks. It looks like I inadvertently broke the pipelines by trying to add a caching step. I've revived them now but I'll try and look at that as a separate issue as the build takes a while now.

**Pull request merged**